server.js

```diff
@@ -1,20 +1,126 @@
 const express = require("express");
 const cors = require("cors");
-
+var bodyParser = require('body-parser');
+const { check, validationResult } = require('express-validator');
 const app = express();
-
+const port =process.env.PORT|| 3002;
 app.use(express.json());
 app.use(cors());
+app.use(bodyParser.urlencoded({ extended: true })); 
 
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
 
-app.get("/", function (request, response) {
-  response.send("Hotel booking server.  Ask for /bookings, etc.");
+app.get('/', function(request, response) {
+  response.sendFile(__dirname + '/index.html');
 });
+// app.get("/", function (request, response) {
+//   response.send("Hotel booking server.  Ask for /bookings, etc.");
+// });
 
 // TODO add your routes and helper functions here
+app.get('/bookings', function(request, response) {
+  
+  response.json(bookings)
+
+});
+//create new booking
+app.post('/bookings',[check('email').isEmail(),],  (req, res) => {
+  const errors = validationResult(req)
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.array() })}
+  let booking={
+    id:bookings.length+1,
+    title:req.body.title ,
+    firstName:req.body.firstName,
+    surname:req.body.surname,
+    email:req.body.email,
+    roomId:Number(req.body.roomId),
+    checkInDate:new Date(req.body.checkInDate),
+    checkOutDate:new Date(req.body.checkOutDate)
+  }
+  
+  console.log(booking.checkOutDate);
+  console.log(booking.checkInDate.toLocaleDateString());
+  if (booking.checkOutDate>booking.checkInDate){
+    booking.checkInDate=booking.checkInDate.toLocaleDateString();
+    booking.checkOutDate=booking.checkOutDate.toLocaleDateString()
+  if (booking.title.length>0 && booking.firstName.length>0 
+    && booking.surname.length >0 && booking.email.length>0
+    && booking.roomId>0
+    && booking.checkInDate.length>0&&booking.checkOutDate.length>0
+    
+   
+    )
+  {
+    
+bookings.push( booking);
+  res.json(bookings);
+}
+  else
+    res.status(400).send(" please enter a value for all inputs")}
+    else
+    res.status(400).send(" please enter a resonable dates") 
+});
```

Comment:
Your checks work well. Ideally, try to avoid nested if statements. You could use express-vaildator to check that the mandatory fields are not empty. I think you could also check the dates using express-validator too, although this is quite advanced and not very elegant, so I think doing it your way is fine. Maybe put it into a separate function to make the code easier to read.